src/client/pages/Offer/index.tsx

```diff
@@ -91,41 +92,49 @@ export const OfferPage = ({
     },
   })
 
-  const bundleData = quoteCartData?.quoteCart.bundle
-  const redeemedCampaign = quoteCartData?.quoteCart.campaign ?? null
-  const checkoutMethod = quoteCartData?.quoteCart?.checkoutMethods[0]
-  const checkoutStatus = quoteCartData?.quoteCart?.checkout?.status
-
-  if (quoteCartError) {
-    throw new Error(`Error fetching quote cart: ${quoteCartId}.`)
-  }
-
-  if (!isLoadingQuoteCart && !bundleData) {
-    throw new Error(
-      `No bundle returned to show offer with (quote cart ID=${quoteCartId}).`,
-    )
-  }
-
-  const bundleVariants = (bundleData?.possibleVariations ?? []) as Array<
-    QuoteBundleVariant
-  >
-
-  const isInsuranceSelectorVisible =
-    isInsuranceToggleEnabled && bundleVariants.length > 1
-
-  const selectedBundleVariant = getBundleVariantFromInsuranceTypesWithFallback(
-    bundleVariants,
-    selectedInsuranceTypes,
-  )
-
   const checkoutMatch = useRouteMatch(
     `${localePathPattern}/new-member/sign/${quoteCartId}`,
   )
-  const toggleCheckout = createToggleCheckout(history, quoteCartId, pathLocale)
 
+  const selectedBundleVariant = getSelectedBundleVariant(
+    quoteCartQueryData,
+    selectedInsuranceTypes,
+  )
   const offerData = selectedBundleVariant
-    ? getOfferData(selectedBundleVariant.bundle)
+    ? getOfferData(selectedBundleVariant?.bundle)
     : null
+  const isReferralCodeUsed =
+    getMonthlyCostDeductionIncentive(quoteCartQueryData) !== undefined
+  const checkoutStatus = getCheckoutStatus(quoteCartQueryData)
+  const checkoutMethod = getCheckoutMethod(quoteCartQueryData)
+  const bundleVariants = getPossibleVariations(quoteCartQueryData)
+  const campaign = getCampaign(quoteCartQueryData)
+
+  useEffect(() => {
+    if (offerData) {
+      trackOfferGTM(EventName.OfferCreated, offerData, isReferralCodeUsed)
+    }
+  }, [offerData, isReferralCodeUsed])
+
+  if (!isQuoteCartEnabled) return <Redirect to={`/${pathLocale}/new-member`} />
+  if (isLoadingQuoteCart) return <LoadingPage loading />
+
+  if (
+    quoteCartError ||
+    !offerData ||
+    !selectedBundleVariant ||
```

Comment:
Could we name these variables differently or add comments?

Like:

```
quoteCartError || offerDataNotFound || isCartWithoutQuotes
```

I can't even figure out what `!checkoutMethod` actually means/indicates hehe

Reply:
Added some functions here. Let me know if you think this is better